src/background/modules/InstallUpgrade.js

```diff
@@ -43,12 +43,14 @@ function handleInstalled(details) {
     case "update":
         if (Notifications.SEND) {
             const [major, minor, patch = 0] = details.previousVersion.split(".").map((x) => Number.parseInt(x, 10));
+            // The autocorrection feature was disabled by default in version 0.5.1
+            const disabled = major === 0 && (minor < 5 || minor === 5 && patch === 0);
             // TODO: This will need to be localized
             browser.notifications.create({
                 type: "basic",
                 iconUrl: browser.runtime.getURL("icons/icon.svg"),
                 title: `✨ ${manifest.name} updated`,
-                message: `The “${manifest.name}” add-on has been updated to version ${manifest.version}. Click to see the release notes.\n\nThe experimental Unicode autocorrection feature ${major === 0 && (minor < 5 || minor === 5 && patch === 0) ? "has been disabled by default" : "was disabled by default in version 0.5.1"}. Open the options/preferences page to reenable.`
+                message: `The “${manifest.name}” add-on has been updated to version ${manifest.version}. Click to see the release notes.\n\nThe experimental Unicode autocorrection feature ${disabled ? "has been disabled by default" : "was disabled by default in version 0.5.1"}. Open the options/preferences page to reenable.`
```

Comment:
Note, usually for that reason _unchanged_ settings are never saved and you can use the default `AddonSettings` file to set the default values. As long as users did not manually modify them, you then get the new defaults.

It's intended exactly for that use case, so as far as I see such a notification should not be needed.

Reply:
I am not sure what you mean. As you know, the Unicode autocorrect feature was previously enabled by default, but we just added a new option in #72 which instead disables it by default. Because a new option was added, the autocorrect feature will be disabled for all users, regardless of any existing settings they may have had. This notification was just to inform the user that the add-on was updated, since Firefox/Thunderbird of course does not notify users when their add-ons are updated.

Considering that the Unicode autocorrection feature is half of this add-on, I thought it would be a good opportunity to also let them know that it was disabled by default and that they would need to manually reenable the feature if they want to be able to continue to use it.

Reply:
Ah no makes sense sorry, missed/forgot that. Thanks!